File: main-backend/src/main/java/com/vvs/mainbackend/pojo/PostDto.java
package com.vvs.mainbackend.pojo;

import java.time.Instant;

import org.bson.types.ObjectId;

import lombok.Data;

@Data
public class PostDto {
  
  private ObjectId id;
  private String _id;
  private String content;
  private String title;
  private String username;
  private Instant createOn;
  private Instant updateOn;

  public void set_id() {
    _id = id.toHexString();
  }

  public String get_id() {
    return _id;
  }
  
}

File: main-frontend/src/app/models/postDto.ts
export class PostDto {

  id?: any;
  _id?: string;
  title: string;
  content: string;
  username: string;

  constructor(title: string, content: string, username: string) {
    this.title = title;
    this.content = content;
    this.username = username;
  }
}